### ISSUE TITLE
Transaction Building/Signing/Verifying Test Vector Brainstorm

### ISSUE THREAD

**Issue by nkohen** [discreetlogcontracts/dlcspecs#47]
I now have a working implementation of adpator-based DLCs. I would like to begin brainstorming a list of cases we want to make sure are included in the transaction building test vectors I will be generating.

Things that come immediately to my mind:
- "Vanilla" construction for n outcomes where n = 2, 3, 5, 8, 100 and various oracle signatures with computed completed CETs (Happy Path)
- "Vanilla" construction with invalid oracle signature
- "Vanilla" construction with valid oracle signature of invalid event
- Invalid refund signature
- Invalid DLEQ proof on one CET, all CETs
- Invalid adaptor signature (with valid DLEQ proof)
- Invalid adaptor signature and invalid DLEQ proof
- Invalid funding signature
- Non segwit funding input
- Invalid final address
- Invalid offer
- Invalid accept
- Sign message missing signatures

Please add to this list and/or critique it!

**Comment by Christewart** [comment]
Closing #30 as this supersedes it

**Comment by benthecarman** [comment]
- Negative outcome values
- Incorrect fee rate on tx
- fee rate given was below network min
- party did not fund the correct amount
- invalid change address
- invalid change amount
- CET has extra inputs
- CET has missing/no inputs
- CET has extra outputs
- CET has missing/no outputs
- funding tx has extra inputs
- funding tx has missing/no inputs
- funding tx has extra outputs
- funding tx has missing/no outputs
- wrong nLocktimes
- wrong sequence
- wrong keys in funding output script
- funding output has wrong script

**Comment by nkohen** [comment]
>     * Negative outcome values
> 
>     * Incorrect fee rate
> 
>     * party did not fund the correct amount
> 
>     * invalid change address

These I like

>     * invalid change amount
> 
>     * CET has extra inputs
> 
>     * CET has missing/no inputs
> 
>     * CET has extra outputs
> 
>     * CET has missing/no outputs
> 
>     * funding tx has extra inputs
> 
>     * funding tx has missing/no inputs
> 
>     * funding tx has extra outputs
> 
>     * funding tx has missing/no outputs
> 
>     * wrong nLocktimes
> 
>     * wrong sequence
> 
>     * wrong keys in funding output script
> 
>     * funding output has wrong script

These I'm a little unsure about as transactions are not actually communicated between parties so these cases will just result in invalid signatures right?

**Comment by benthecarman** [comment]
>These I'm a little unsure about as transactions are not actually communicated between parties so these cases will just result in invalid signatures right?

Yeah, I guess these are more things to that people should testing with the test vectors

**Comment by nkohen** [comment]
Good point, I guess I could generate a bunch of invalid signatures in this way with error messages about how they were generated to make life easier for people when debugging

**Comment by nkohen** [comment]
Funding transactions with different kinds of exotic change output SPKs

**Comment by nkohen** [comment]
Funding transactions with various numbers of inputs from each party
